### ISSUE TITLE
Thread about potential improvements for the Introduction to Julia course on JuliaAcademy.com

### ISSUE THREAD

**Issue by logankilpatrick** [JuliaComputing/JuliaAcademyMaterials#23]
Thread created during Google Code-In for feedback.

**Comment by ashimantha** [comment]
learning with Julia is not bad. that is the easiest method to learning. we can learn correctly through the video guide.
![Picture1](https://user-images.githubusercontent.com/59006753/71553745-be748b00-2a3a-11ea-9466-a6d4195c17a9.png)

**Comment by mehardeep88** [comment]
Overall the course was great and beginner friendly. There should have been more quizzes and exercises too.

**Comment by soumitradev** [comment]
The course was well explained in general, but here are a few things I noticed:
- In the “Review” quiz under basics, the 4th question needs to be updated. The answer says that `push!` accepts only a single argument to push to the `Array`, while in reality, `push!` accepts multiple arguments.
- The Quiz questions could be randomized since retaking gives the same set of questions in the same order.
- The course slowly becomes lesser interactive towards the end. Maybe some quizzes in each section will keep the students engaged?
- The “Advanced topics” section (especially Linear Algebra and Factorization videos) should not be quizzed, or even required for completion of course since most high school students (including me) will have no idea what’s going on for a solid 20 minutes. The videos include much higher-level linear algebra as compared to the most advanced AP exams. Though high schoolers know about matrices, determinants and some matrix operations, they don’t have much knowledge about the topics covered in those videos.
- The progress bar bug (mentioned in #22) prevails and appears to be a sitewide issue.
- Maybe coding exercises (ones that include writing a script) can be implemented, where the script is run in a `venv` in the server, and its output is validated for various test cases.

**Comment by josobar** [comment]
The course is pretty good overall. I was able to understand everything up to the linear algebra section. I can agree with @soumitradev on many points including adjusting the volume of the speaker and perhaps including more interactive activities to keep the viewer engaged.

**Comment by Saif1791** [comment]
In the “Review” quiz under basics, the 4th question needs to be updated. The answer says that push! accepts only a single argument to push to the Array, while in reality, push! accepts multiple arguments.
The Quiz questions could be randomized since retaking gives the same set of questions in the same order.
The course slowly becomes lesser interactive towards the end. Maybe some quizzes in each section will keep the students engaged?

**Comment by mikolajhojda** [comment]
I think should be more review tests and maybe in another form. Checking our knowledge gives a lot.
Overall the course is nice.

**Comment by TheComputerM** [comment]
In my opinion, there should be more quizzes to keep the learners engaged and the lectures should be more interactive. There should be more exercises and interactive activities in each section to make it more interesting. It is also a good idea to keep track of the learner's performance in each activity and give the user a score or a grade based on his performance. Overall the course is very informative and is great for beginners.
![Intro to Julia](https://user-images.githubusercontent.com/24741538/71590126-59f62080-2b4d-11ea-8c21-238fecd69639.PNG)

**Comment by aftex261** [comment]
1.Overall, If I have to rate the course then i would rate it 4.1/5. The explanation of prior topics related to strings,data structures was good.
2.Learning Julia is overall a good combination of getting accuracy like C programming language and less lines of code(LOC) like Python. 
3.Julia can beat python in terms of numerical analysis and data analytics .
4.Going to the end,The last lectures on basic linear algebra and factorization was good but lectures were not that interactive and amazing.
5.At last,I would like to say that my learning experience was very nice but minor improvements during the course prerequisite should be done.
Thank you

**Comment by danieldzou** [comment]
I'm a little new to programming, I had some experience with Java, but never really got into C++ or FORTRAN, and I am completely new to Julia, but I love it so far.

The introductory course was the perfect way to introduce someone to Julia with programming experience! It might have been a little face-paced, but I love how you can just access the resources in the Jupyter Notebook to go back and try it yourself.

I liked the first review, but was hoping for more review quizzes to check and assess your understanding.

All in all, I think it's a great introduction to Julia and showed me more about the language and what it was about.

**Comment by Ninad10code** [comment]
I knew about programming languages like C, C++, Java, and python, hence things went smooth at the start till functions. But from packages and plotting, it got a bit of trickier to understand. Maybe a longer video will help clear the concept more effectively. Overall it felt really amazing learning a new language Julia. Thank you.

**Comment by minukag** [comment]
The course and the content in it is really great. I am very happy with learning julia. I Learned lots of cool tricks.

Here are some improvements I think could be implemented:

-  Voice was mostly clear but I would like if closed captions were available.

- For exercises: after each section, the exercises like https://www.freecodecamp.org/ would be much better.

- links to some sample programs made with the techniques taught will be great.

- Add labels in video progress bar to subtopics in the video(i.e: data structures -> dictionaries, tuplets, arrays).

- I think it would be helpful if it was 2 different courses as basic and advanced.

**Comment by Wingfwd644** [comment]
As someone who is relatively new with programming (have some knowledge with C++ and Java), this course was a good introduction to the world of Julia. Personally, the course in general was quite fast paced, and I had to pause the videos a few times(more so at 'packages and plotting' and 'advanced topics') to understand. Maybe you could add a few more videos, especially at the 'advanced topics'? Like for example, factorization is a longer topic so you could break it up into 2 videos. For me I started to get lost at the basic linear algebra.

Also, kind of much like everyone else has already said, there needs to be more reviews or interactive quizzes. Its useful to see whether we understand the material or not. Adding them at the end of each section, but then at the advanced topics section maybe just add them at the end of each topic. Also, other than multiple choice you could also add the ones https://www.freecodecamp.org/  does(or basically those ones that lets you interact with the code.

Also not a big deal, but at my school we haven't been thought stuff like LU factorization(I'm technically still in junior high) , so maybe a brief introduction to those before just going straight into it would be good.

But yeah, overall I enjoyed learning about Julia, it was an amazing experience. 9.5/10

**Comment by noisya55616** [comment]
I think that this course was a great introduction to the basics of Julia. Everything was pretty understandable. I did notice a lack of many review quizzes: there was only one, and so most of the course was watching and typing along. Perhaps some more quizzes or similar things for a bit more engagement could be in order. I also noticed that next.juliabox.com, which the instructor recommended in order to follow along with the examples in the videos, no longer worked. It would appear that JuliaBox no longer provides any free services outside of a 30-day free trial (https://discourse.julialang.org/t/julias-growth-is-making-free-juliabox-unsustainable/19441). Perhaps this should be removed and replaced with an alternative.

**Comment by abhijeetsinghcode** [comment]
The course was good but It was not very interactive.The course should be like the Mentor  writing the code not  Mentor only scrolling content  and explaining . Although she was explaining greatly but for beginners if he/she don't know about coding when he/she will writing the code he/she may come across error like for example if we take  about python there is irritating error of Indentation which irritating to good programmer also so if she was writing the code and students were following along with her it will great practice to not face any error or problems and students mind also feel good with Julia codes

**Comment by theEpicOne1** [comment]
•	There were a few parts where I noticed relatively essential functionality that was not directly (or at all) addressed. For example, how do you iterate by a number other than 1 in a for-loop? I did not see this covered. 
•	I found the instructor for the introduction course to be very engaging and personable. She adds in interesting “Easter-eggs” like certain variable names which lighten up learning the course.  However, it seems as if the "Review" quiz was an afterthought since there is only one at the beginning and not after the longer videos. These would benefit more a quiz to make them remain engaging than the more quick and simple concepts towards the start which are digestible on their own. 
•	If I return to an older tutorial to check something for myself, it makes the “finished” check mark icon turn back into the “in-progress” half circle. Once you leave the video again it returns to “finished.” This seems unnecessary and confusing to the user since they already had to confirm “complete and continue” to get the check mark. Making them wonder if they must do it again for completion is unnecessary. It would be best for it to just stay a check mark forever once they have hit “complete and continue.”
A side note on the language itself; I absolutely love the ability to declare a function as mutating/non-mutating. This is a lot easier than having to deal with unintentional aliases in a language like java (my first and current language).

**Comment by chinglamchoi** [comment]
**Positives:**

_1. Very interactive and easy to follow (except the last lecture)_
I really appreciate how the instructor chose to use IPython notebooks to teach instead of jl files, and provided the code for each lecture for reference. This makes the course very easy to follow along with, especially for those who are uncomfortable with command line interfaces.

_2. Very detailed and clear_
Every notebook is annotated very clearly with markdown. This is especially helpful for later lectures which cover linear algebra: the latex explains the code very well, making it possible for complete beginners to understand. Also, I really like how the instructor compares Julia and Python, differences between different data structures (tuple vs array), effects of different methods of declaring strings – this fine-grained teaching style really allows students to gain sound understanding of Julia and its unique advantages.

_3. Others_
I really enjoyed the generous amount of maths integrated into the course, as well as how she eased us into more advanced topics like plotting and linear algebra with Julia.


**Less-positives**

_1. More quizzes please_
The short quiz was useful in consolidating knowledge, it would be even better if more quizzes were included in the course after each section.

_2. Variable scopes_
Although the explanations were very detailed overall, we could have used more elaboration, especially on the scope of variables in the loops & conditionals & functions section. People who learned e.g. Python first will likely have some trouble adjusting to Julia variable scopes.

_3. Difficulty of last lecture_
I think the last lecture was a little too advanced to be included in an introductory course. Perhaps it could be allocated to another course which focuses on linear algebra/calculus with Julia.

**Comment by kchatr** [comment]
I enjoyed the course and felt that it covered the essential foundations of the Julia programming language. The overall UI and UX are well-designed, thoughtful, and intuitive. I like the minimalistic design of the site and the pragmatic look and feel to the course.

Maneuvering the courseware was no issue and returning to my current point was fast and efficient. I do wish that there were captions, however, not because it was difficult to hear or understand, but because reading allows viewing the course in a more flexible manner.

I do, however, wish that it went into a little more detail about the benefits and vast uses of Julia, as well as what its primary goal as a language is. Also comparing and contrasting it to Python, a very similar language syntactically, would be helpful.

The course material is well designed and is structured in a logical and easy-to-follow manner, but I feel it could go a little deeper into the content. It is great at providing a solid base, but I don't feel it goes into enough depth or breadth for one to use it for data analytics and advanced mathematics. Going further into the implications and applications of mutating and broadcasting functions, the ternary operator, and other more advanced topics would be educational and helpful. The final two lectures I found were rather complex and assumed an in-depth knowledge of matrixes and their operations. For benchmarking, it did not explain the syntax or how importing C/Python code is fully accomplished. 
![Screenshot (12)](https://user-images.githubusercontent.com/31332999/71648473-97d08180-2cd2-11ea-8d74-000caf4b38b6.png)

It would also be helpful to cover the topics with a holistic view. For instance, explaining the range notation and how you can state the magnitude of the steps (0 : 2 : 10 - generates all even numbers from 0-10 inclusive). Also, for the plotting section there was very little information on how to structure a plot, as well as what exactly plot and scatter! do. At times, the material can feel a little rushed and overfitted.

Also, they talk about the '!' used in functions after the quiz (see screenshot), so it does not make sense testing content that has not been covered and is open to interpretation
![Screenshot (13)](https://user-images.githubusercontent.com/31332999/71648479-af0f6f00-2cd2-11ea-9904-74996a1f84bc.png)

More quizzes and assessments would also prove beneficial, and assist in cementing ones knowledge.

**Comment by ishita272001** [comment]
Although it was a good course.
1 My favourite course among this was loop.
2.Mainly it explains everything in detail almost.
But as one is perfect, I found some issues working with it.
1.No Tests-There was only one test and that only with five questions. More tests should be added.
![Quiz](https://user-images.githubusercontent.com/59233460/71632450-a700f200-2c34-11ea-98fd-2ed52b3913e2.PNG)
2.Volume in some courses was less. I used earphones for that.
3.In some phases video loading was slow (although it can be my fault also)
4.Some level of course should be decreased as it is little bit of advance level.
Hope you work on my issues and make improvements.
THANKS

**Comment by Redeem-Grimm-Satoshi** [comment]
First of all, i will like to say this is a great initiative to create a learning platform like this, so newbies in Julia programming will find it easy to get started with the language without having to go through documentations, do lots of google search and spend time on YouTube looking for tutorial videos. I am new to this language but i am an experienced Java developer and i just finished the Introduction To Julia Course. And **it help me learn some core concepts in Julia** but during the learning process, i found some issues that if sorted out will greatly **improve user experience.** And most of the issues i experienced are mentioned above(volume is not loud,quiz is not enough, lesson not interactive issues) but will bring out a new one.

### First Issue:
When about to take a course i didn't see the course Objectives and Requirements but only saw a little bit of description about the course. I think if the course objectives, a solid description, requirements is provided, then user experience will be improved.

### Second Issue:
When i completed intro to julia course, the progress bar shows 0% complete
![Screenshot_2020-01-03 JuliaAcademy](https://user-images.githubusercontent.com/45304978/71706053-f127d600-2de2-11ea-9ec5-0946a3b4ef1c.png). If issue fixed then user experience will definitely aggravate.

### Last Issue:
Not everybody(especially me) have access online on daily basis, I think the instructor would have used the Julia REPL or in a text editor like vscode, vim,sublime,etc.. not jupyter notebook so that users really get to see the work flow and problems encountered when installing packages(especially plots.jl) on julia repl and using them. I think if this issues is fixed, then users will find it easy to follow the course and try out examples easily on their local computer. Thanks

**Comment by SPARTACUS5329** [comment]
The course is good(overall). I(Aditya Kovilur) think:

1.Although it is just an introduction , there should be a mini-project in the end the instructor shows how to create to actually make the viewer understand how all the parts work together.
2.The course should include the reasons why Julia is better than other languages(if it is, because I think it is just a derivative of Python) and it's scopes .
3.There should be a lesson after the intro which gives examples how Julia can be integrated with other languages to create something extraordinary(like Python with R and SQL for Data Analysis)

**Comment by trizalb** [comment]
> Thread created during Google Code-In for feedback.

i loved the intro video it was quick,amazing,innovative , and easy to learn
this is great for new learners who want to learn this .
i don't think it require any improvement except it is boring,you should make it more intresting

**Comment by Anikamittal23** [comment]
i really learned  a lot from the " introduction to julia" course .some things i found difficult but when i will work more on julia i will be more familer to it.

- Their should be more quizes

- Volume in some courses was less. I used earphones for that.

**Comment by 22dileeps** [comment]
I personally like the way the course was set up which made it easier to understand the easier concept and the hard ones. The problem was that there was not many practice question like the Basics. Normally intro courses have multiple practice questions or sample projects to do to get going, but this intro course had only one review and no sample projects. For me, the sound quality of the video was a bit of issue, but overall I think that this was a great intro course for newbies like me.

**Comment by anshCoder** [comment]
Good Courses Offered By Them :)
* Good explanation, Quick to Understand!
* Quiz is Really Awsome Based on Concept!

My overall Review is (9.5/10)

**Comment by aasangamnerkar** [comment]
The Course was easy to follow and moved at an appropriate rate, but there are a few things I would change to improve it further. 
1. Instead of cutting the entire lecture into different lessons, either have the option to view the entire video in one sitting, or have different videos for each section. The choppiness where the video was cut interrupted the learning flow for me.
2. Involve the student more. I feel that adding more questions on the material covered in the lecture would prove beneficial to the retaining of the new skills learned in this lecture.
3. Consolidate all the information into notes as well. I would have been able to learn faster if I could read along with a powerpoint or written notes as the video went on.
![proof](https://user-images.githubusercontent.com/36603983/71786655-49bdc580-2fd3-11ea-8a7f-041ea16689ac.PNG)

**Comment by jatin2003** [comment]
The Course was overall great and informative for beginners in Julia. But some improvements could be made in " Introduction to Julia" Course :- 

- There was only one test with five questions only. I think more Quizzes should be added at end of every video.

- Although Voice was clear, Closed Captions can be added for more better understanding.

**Comment by sankalpdass** [comment]
FEEDBACK

![PhotoEditor_20200105_122849529](https://user-images.githubusercontent.com/56037184/71793878-d795c100-3064-11ea-8d5c-33f616ade09e.jpg)
![PhotoEditor_20200105_160611110](https://user-images.githubusercontent.com/56037184/71793886-e0869280-3064-11ea-867d-fac343d3a75b.jpg)


Actually the course on Julia academy are so interesting that I watched all videos of your forum including "introduction to julia" course.
Now I observed some pros and cons of your course i.e. "introduction to julia" and some details are mentioned below : -

Pros:-
• tutorial video of course teach about how to code with Julia language.

• content of topic description at starting of each video was very helpful.

• the concepts taught on the video was very helpful to solve basic problems.

Cons:- 

(The cons are noticable. You must apply these changes to make it more better.)

• The starting of course is not much interesting because the instructor do not tell the advantages to learn Julia language at starting of course but later on she described it but, as we all know first impression is the last impression so she must explain the advantage at the very beginning of course to make the course interactive for Julia learner.

• there is no video of installation of julia in IDE, how to get friendly with this Julia language and what else we can do with this in future.

• most important there are no minor projects covered in the video tutorial as we all know without project we can't understand any language.

• also the video are too short to understand because functions can't be understand in just 10 minutes.it takes time to understand function.

• The instructor of video teach very fast and skipped many concepts that must be taught to understand any language to learn coding. For example she didn't give many examples to understand loops at first time. 

[combinepdf.pdf](https://github.com/JuliaComputing/JuliaAcademyMaterials/files/4024651/combinepdf.pdf)

**Comment by SwayamGos** [comment]
The julia intro course was a really good course, to begin with. It started with the basics and went all the way to loops and linear algebra which is commendable. However I would like to see a few improvements:
1. There should be more focus on as to how much the student is actually learning. I completed the entire course and I found just one quiz that too in the very beginning after just 4 lessons. There was no quiz in the end of the course too.
2. I would've liked if more examples were included in the video to show looping structures. Loops can be of many different types and showing more examples by tweaking small stuff here and there would have been better.
3. Also the quizzes should have their questions changed every time you take the quiz. I took the quiz once where I scored 4/5. I retook it. The same questions. I easily got 5/5. It'd been better if the questions were differently set from the same topics so they could check my actual knowledge.

**Comment by ABHI-310** [comment]
The Introduction to Julia Course was very interesting, well I fortunately toke this course first and then the Machine Learning course with Knet (Using Julia Language) . Both the courses were very engaging.
Introduction to Julia was really good, It introduced me to a new language, I found it very similar to python.
The thing that I would like to appreciate the most was that the mentor was so good in her lectures. She had a rich knowledge of the subject. Another appreciable thing was the Quiz, I scored 4/5 in my first attempt and a perfect 5 in second attempt. 
Another that I would recommend is that to add similar quizzes after each lecture. That help students to have a brush up of the things learned in the course. 
All in all it was a nice introductory course and would rate it 8/10. : )

**Comment by Dhanus3133** [comment]
1. To add Dark mode feature in the Settings

2. After completion of courses, Certificates can be auto-generated

3. To add a quiz after the completion of the concept in the video, it may be an MCQ and it may even evaluate the percentage scored in the certificate

And rest of the track was good

**Comment by Miyuki-L** [comment]
1. The course is structured nicely into different components but I think that there could be more reading checkpoints or quizzes that make sure that the learner follows and understands the contents. Possibly also including quizzes or exercises that involve coding or identifying the input and/or output of some code provided. 
2. Depending on who the course is targeted at the content may be confusing for the learner. Some content-specific terms, like index and left division, could be explained more if the course is targeted at people with little to none coding experience and/or people who have not learned Linear Algebra and matrices before. When the course starts to use matrices and operations with matrices, it will get confusing to the learner if they do not have a sufficient understanding of matrices.
3. It may just be me but the link provided for the Jupiter notebooks did not work, so it made following along with the notebooks not possible.

**Comment by fungktt** [comment]
Hi! I believe that this is a great course, for several reasons:
- It was very beginner friendly and was suitable for anyone without prior programming experience (like me) to get started
- Taught me to use Jupyter Notebook - a very powerful tool
- The notebooks were very detailed and contained loads of explanations with markdowns which became very helpful in consolidating the knowledge I got from the lectures earlier on

Some improvements
- More quiz to solidify the knowledge learnt in the course
- The quiz should be made optional for those who would like to do them, but not everyone, because quizzes in the LInear Algebra section is very advanced and is not suitable for everyone
- JuliaBox is no longer free so that should be updated
- I saw more Jupyter Notebooks for this course on the GitHub repo, so perhaps the links for those can be added on this course as well?
- The pace got faster too quickly and it may be too fast for some beginners to understand (especially towards the latter end)
- 1-based indexing can be made a little bit clearer and some people don’t know what it means - also it could confuse people coming from Python (which is 0-indexed)
- The quizzes should be longer and changed every time because anyone could easily retake it and get full marks the second time round
- It would also be good if we could have a summary page/video after each chapter/section to go over what was covered
- Remove Jane Herriman's **personal email address** from the Strings lecture (I think this needs to be blurred out because she needs privacy)

<img width="1419" alt="img" src="https://user-images.githubusercontent.com/28962234/71900304-7863a980-3155-11ea-8f03-4d2eef15efc7.png">

**Comment by nimtara** [comment]
Julia Academy gives us a great knowledge about computer programming languages. It enables us to learn coding activities such as strings, loops and tables, fractions and many more. Moreover the programmes are clearly shown in each page with clear videos and screenshots. The tasks enable beginners to cooperate efficiently no matter the person is clever in programming orelse not. The videos are very clear and helpful. On the other hand I request you all to add more quizzes at the end to make the user enthusiastic.
Good luck for Julia academy!

**Comment by skalyan22** [comment]
I enjoyed the introductory course, for the instructor was very knowledgeable and was good at explaining concepts. One major thing I would have liked to see is that since it is an introductory course, some concepts I only understood because I have learnt another programming language in the past. I feel as if some of the lectures should be broken down even further to explain the syntax a little bit more because the course is an introductory course. The lecture on loops introduced new syntax very quickly and I would have preferred if the lecture was slower and explained each part thoroughly before moving on. In addition, I was unable to reach the next.juliabox.com website, but I had jupyter notebook installed so I could still follow with the video. Alternative ways to access the files should be included in the beginning of the course. I also think that the lecturer should encourage us to test out our own code after learning the syntax so we can better understand the language. Finally, a small, simple project could be added at the end of the course based on what we have learned so far. By doing so, we can put together all the topics we learnt throughout the course, and it would be a fun way to end the course.

**Comment by IamAhsanMani** [comment]
I like the auto generated Cerificate man love that.

**Comment by arby37-tech** [comment]
1.Julia Academy is provided with the 
  correct way of learning Julia.They help.   
  to understand the and learn Julia. 
  Moreover the programmes are clearly.  
  shown in each page with clear videos.   
  and screenshots. 
2.Questions in the quiz could be.  
   randomized since retaking gives the.  
   same set of questions in the same.   
   order.
3.One of the minute default I found is that the mentors did nt write and explain the coding they only scroll and explain. 
4.I also feel that adding more questions on the material covered in the lecture would prove beneficial to the retaining of the new skills learned in this lecture.
     This is all what I found from my point of view

**Comment by Airiinnn** [comment]
**Positives**
1. Videos from the first 4 sections (Introduction - Packages and Plotting) were very easy to follow and understand.
2. The notebooks provided were really useful for us to follow along and try with different inputs.

**Possible areas of improvement**
1. It would be nice to have more quizzes or maybe simple problems for us to solve using Julia for us to gain more hands-on experience learning Julia.
2. The Advanced Topics section was hard to follow as many of the terms were not explain but merely stated. This could be due to the fact that I am quite new to programming, hence the lack of knowledge of certain terms.
3. JuliaBox also didn't work for me despite trying with my GitHub and Google accounts. Had do everything locally which was fine.

Overall a good course for picking up the basics of Julia, maybe leave the advanced topics to another intermediate course.

**Comment by IamAhsanMani** [comment]
1) The video is not loading correctly it should show video according to the user network coverage.

2) They should not increase the complete progress bar unless user watch video beacuse user just click one continue to next lesson without watching video and increase the progress bar.

**Comment by peaceradiant** [comment]
![JuliaAcademy_100%](https://user-images.githubusercontent.com/46359762/72007811-30c74580-325b-11ea-8f34-78c7714f9476.png)
The course was good overall. The concepts were explained in a good manner, so even unexperienced programmers could understand. I recommend adding more examples in the future for every new feature presented. Also, I wish that deeper aspects like arrays, 2D arrays, functions etc to be even better explained (those are basic concepts used a lot in programming). I loved the pointed similarities between Julia and other programming languages like Python, C/C++ and JavaScript. I highly recommend this tutorial for those who are already experienced with some programming visions and techniques so it can be even easier to understand. I would also like to see the tutorial being filmed like it is more real(the person being shown standing at the computer, being filmed from multiple angles etc), instead of just a squarish camera attached (it is just my opinion, others may find it more fun this way). I was satisfied with the user experience, the progress bar was updated along the way but I was a little disappointed to see just a little quiz at the beginning (I highly recommend adding more of those, those are essential). A dark/night switch mode would be great, especially when big concepts like factorizations are being showed.
Again, the course is great, Julia is great and I hope that some of my recommendations, along with others to be implemented in future tutorials/videos. Keep up the good work!

**Comment by DaniyalHayat** [comment]
![Screenshot (18)](https://user-images.githubusercontent.com/32670352/72034490-ba2a4800-325a-11ea-94c3-7fe0de4cdb74.png)
The course was great and informative. This course could have been better if it had more review sections such as the first one in this course to keep the concepts we have learned much better solidified in our memory. For example, a review after every 2 videos would have been great.

 Another way I would have liked the course better if instead of already pre-built code, the instructor would have written the code step by step. This way, personally to me, could have been a better way to absorb all the materials.

**Comment by NandVinchhi** [comment]
The Intro to Julia Academy was a fantastic course and I truly enjoyed it for the following reasons:

> I loved the fact that they included video tutorials. That makes it so much easier. Everything was explained to the point and extremely well!

> Short and Sweet: All the information was concise and the videos were short. The information was conveyed in such a way that it was very easy to understand

> content: The curriculum is extremely well designed and it contains all the basic and necessary tutorials to Get into Julia Programming.

Suggestive improvements:

> Add more quizzes. I feel that there should be a quiz every 1 or 2 videos. 

> If you want to retake a particular quiz, it shows you the same questions in the same order with the same options. I would suggest making the quizzes more dynamic so that there is some variation in the quiz when you retake it. Maybe more emphasis on the questions you got wrong last time?

> If this course is targeted towards people who have never coded before, then I feel that it could be made a little less challenging or a little slower. The advanced topics required knowledge of some relatively difficult math concepts such as matrices. I think these concepts should be explained a little in the videos. 

> I would also like a system where you have to complete assignments and submit code. This assignment can be tested online by the website and the required feedback be given. Maybe something like codecademy?

**Comment by Aditya-Harsh** [comment]
The instructor truly knows how to explain and get the student to understand. The course also shows that the instructor put in a lot of time and effort into making it.Since it required no previous coding experience, I suggest it for complete beginners.
However the lecturer has to add a section that express the basic differences in the new version of Julia I guess.
I expect the to add subtitles to the video.
Also , I advise them to remove facecam at corner in video to make learning experience much better.

**Comment by D3nii** [comment]
Okay, the course was super awesome and all I would possibly want is for it to have some kind of tests in it. Overall, it was way better than "Foundations of Machine Learning."

**Comment by D3nii** [comment]
Trying the second time, the website is a bit broken, when we revisit the course landing page, the progress bar is 0% when I've already done all the course, it must work like if I go to the course after completion, the certificate should show up immediately. Secondly, More quizzes, The linear algebra part is a must quiz because those concepts must be tested further, you guys aint issuing certificates to random people who don't get what is in the videos, do you? and just like freecodecamp.com, they offer a place to code live and to progress, I'd love a similar approach. Thats all.

**Comment by eahmed123** [comment]
> Thread created during Google Code-In for feedback.

The course over all is very good. I really love this course but the quality of the video and audio was not really good. Even the way of teaching was not very perfect and professional. However this course was very informative.

**Comment by Aaron-glitch-ass** [comment]
![Screenshot_2020-01-10-10-11-07-61](https://user-images.githubusercontent.com/58646959/72122062-c36cff00-3397-11ea-9cef-9f23b4fdaca6.png)

The certificate doesn't has a signature, which makes it less accountable.

However, I found out that the lesson's interface is improving, such as the comment section. The explanation is also straight forward and entertaining.

Keep up the good work.

**Comment by saikishore222** [comment]
okay, the overall course was nice,i learnt something new today. thank for giving the free opputurity to learn

**Comment by kobulwidodo** [comment]
Very good class, very helpful for students and also easy to understand

**Comment by kobulwidodo** [comment]
a good reference place for programmers to continue developing their programming skills and what's very important is that it's free !

**Comment by Cracked24** [comment]
I want to start by letting everyone know that Julia Academy is a really user-friendly and easy to use platform. It is free and is very helpful. But nothing is perfect. It could use some of the following improvements:
1) I felt that the introductory course was a little rushed and it would help a lot if the course was a little longer.
2) The course is interesting but it could be more interesting if you add some fun activities and quizzes.

**Comment by eahmed123** [comment]
In the course the teacher did not provide the information in deatils and try to roll the things up. She was in hurry. I didn't understand things clearly. Secondly, she didn't use Technical words regarding the profession. It was not on just one lesson but in many of them So I cannot give reference. I hope you understand what I want to say by the sentence that course was not professional. I cannot detail anymore I guess.

**Comment by eahmed123** [comment]
Secondly, I should say the video was not very clear and in good HD quality and even audio was not very standard. That was the reason I used the word that this course was not professional. Please improve these things, otherwise all things are perfect.

**Comment by chlxry** [comment]
The user interface of JuliaAcademy's website is intuitive and, in my personal opinion, the epitome of functional minimalism. Overall, the lectures were clear and the concepts were well-explained. The visual aid and examples were also very useful in helping me better understand how the language works. One suggestion I would give is to include more exercises (e.g. one at the end of each chapter or checkpoint) to check the student's understanding. Another small improvement that can be made would be to add subtitles to the videos to make learning the language more inclusive. Other than that, the course was perfect for a basic introduction to Julia. Learning the language was very enjoyable :)

**Comment by MinJiaTse** [comment]
it can be improved by having more quizzes throughout the course. I will also like if she could introduce some compiler other than a notebook. Maybe also add in what is Julia used for. However, the tutorials were engaging and made me learn more about Julia.

**Comment by kobulwidodo** [comment]
a good place to develop your programmer skills, the video is very good, the explanation is very easy to understand, the questions given after the video are also very good points, but only a little, it's better to add more so it is more interesting, and
 the last, the class is free which makes it easy for all people to access and learn it

**Comment by eahmed123** [comment]
The was course was awesome and it gave me the good introduction to the julia programming language. It gave me a chance to gain the knowledge and information about this great programming language without even paying a cent. I really love this thing. Secondly, I have seen one thing that should be improved are the practice quizzes. I think if it would be there so the student will need to review things again and again, like this the student can gain more knowledge overall. I am also going to attach screen shot that there was only one quiz, in whole course. I think they should be increased to two or three or may be four, like this learner will also enjoy and will gain knowledge too.
![JuliaAcademy](https://user-images.githubusercontent.com/59716131/72206033-69cc0980-3457-11ea-9e48-a93b9cd68d02.png)

**Comment by eahmed123** [comment]
Other than this everything was perfect and good. I will retake this course so I can review thiongs and again and gain knowledge which was left from taking first time. I will also review and complete other courses from this website, because I really love this. I have completed the first one Introduction to Julia. Honestly, I love this. 
![Certificate of Completion for Introduction to Julia   JuliaAcademy](https://user-images.githubusercontent.com/59716131/72206096-d9da8f80-3457-11ea-9d2a-33c72bc465de.png)

**Comment by shresth2001** [comment]
Review of 'Introduction to Julia' course:
1. More reviews or quizzes should be added.
2. The videos in the starting were really good but as it approached the end (the advanced topics section), the videos became less interactive and often very boring.
3. The only quiz present has to be re-created in such a way that different questions are asked on retrying.
4. More examples regarding the usage of different functions and packages need to be added to facilitate better understanding.
Thank you,
Shresth Bindal.
gci: Shresth2001
github: shresth2001

**Comment by nika02** [comment]
For the "Functions" section, what an "anonymous function" could have been explained. The section could've also touched on why there's different ways to declare a function and when to use which way (if applicable). I liked that the functions got linked back to data structures. That way, watchers can see the relevance of the sections as a collective whole rather than as individuals.

For the "Plotting" section, the "!" in "scatter!" could've been explained with looking at what would happen if "scatter" was used instead.

Although this isn't relevant to the main focus in any section, some code has "$" that appeared across a couple sections. Maybe a brief sentence can be added to explain that since that might be harder to understand if the watcher doesn't have any coding experience.

I couldn't understand the transposition sub-section in "Basic Linear Algebra" and the sub-sections afterwards. The "Factorizations" section was also very difficult to understand. Granted that they are under "Advanced Topics", I still do think that since this is under the introductory course, the math behind it could be explained better and broken down to even smaller bite-sized pieces (especially the Factorizations video since that was a 20 minute long video).

More quizzes or reviews can also be added to keep track of the watcher's progress and understanding, as well as to keep them engaged and interacting with the material.

**Comment by DIVYA1230** [comment]
the issues are that 1st the quiz have same quest

**Comment by DIVYA1230** [comment]
the issues are that 1st- the description could have been more informative like giving information about what we would be doing, etc.
2nd-in the quis even after we re upload i we get the same questions . If the questions vary i woul be more interesting.

other than these i don't find any other issues and also the courses are really helpful and cover important parts.

**Comment by D-Studios** [comment]
I enjoyed participating in this course but I did have a few issues with the videos.

Issues:

    1) The audio sometimes in the videos was not clean. There was background noises that were pretty clear to hear and this was distracting.

    2) I was not quizzed a lot through the whole course. This made it hard for me to retain details which I had just learned.

    3) Finally, I believe either the course should state that there are prerequisites, or cover essential information that is needed to be known before enrollment. For the last two videos of this course, I thoroughly did not understand much due to the fact that I have not learned such complex mathematics yet.

**Comment by benmcclintock34** [comment]
The course was great and educational. It helped me, a beginner coder, learn some of the foundation of the Julia software. For example, I learned about some of the complex math that can be done inside of Julia.

 However, I did notice a few issues.
1.The audio in the videos was not very loud. Even with my computer being on high volume, I could not hear instructor very well during some of the videos.
2. Some of the courses at the end were very challenging and above a beginner level. I am a beginner coder and I understood the first 75% of the courses pretty well. Then I got to the last 4 topics and they were very confusing.
3. There was only 1 quiz throughout the entirety of the lesson. I forgot some of the information I learned because there were no quizzes to help me remember the material. The course should have quiz after every 2-3 videos.
4. Some of the material wasn't explained very well on what it did. For example, in the packages section, they only explained how to use the packages, and not what a package actually was. I was confused on what a package was until later in the video.
